keel-core/src/main/kotlin/com/netflix/spinnaker/keel/memory/MemoryIntentActivityRepository.kt

```diff
@@ -23,25 +23,57 @@ class MemoryIntentActivityRepository : IntentActivityRepository {
 
   private val log = LoggerFactory.getLogger(javaClass)
 
-  private val orchestrations: MutableMap<String, MutableList<String>> = mutableMapOf()
+  private val currentOrchestrations: MutableMap<String, MutableSet<String>> = mutableMapOf()
```

Comment:
I haven't looked specifically at how IntentActivityRepository gets used but I suspect these should be ConcurrentHashMaps

Reply:
Changed.